tritium-lib/src/main/java/com/palantir/tritium/io/InstrumentedStreams.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.io;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public final class InstrumentedStreams {
+    private InstrumentedStreams() {}
+
+    /**
+     * Instruments the provided stream to provide a meter tracking bytes read,
+     * and histogram tracking bytes per second.
+     *
+     * @param in input
+     * @param bytes bytes read
+     * @param throughput bytes read per second
+     * @return instrumented input stream
+     */
+    public static InputStream input(InputStream in, Meter bytes, Histogram throughput) {
```

Comment:
Thoughts on taking a `TaggedMetricRegistry` + name, and using metric-schema to define a standard structure? That way we can define reusable dashboards.

Reply:
Yep, will do when I have some cycles

Reply:
Interested in thoughts for the tagging structure. Right now I have a single `type` tag that one sets to distinguish streams. We will need to be cautious with tag cardinality, though we can enforce compile time tagging.

Example from https://17129-66020851-gh.circle-artifacts.com/0/~/artifacts/junit/tritium-lib/test/classes/com.palantir.tritium.io.InstrumentedStreamsTest.html

```
io.stream.read:{libraryName=tritium, libraryVersion=unknown, type=test-in}
             count = 2147483648
         mean rate = 146651358.56 events/second
     1-minute rate = 155940259.30 events/second
     5-minute rate = 156866618.01 events/second
    15-minute rate = 157027104.69 events/second
io.stream.write:{libraryName=tritium, libraryVersion=unknown, type=gzip-out}
             count = 2147483648
         mean rate = 146654007.80 events/second
     1-minute rate = 155943552.02 events/second
     5-minute rate = 156869248.25 events/second
    15-minute rate = 157029620.15 events/second
io.stream.write:{libraryName=tritium, libraryVersion=unknown, type=raw-out}
             count = 10409991
         mean rate = 710861.07 events/second
     1-minute rate = 752956.85 events/second
     5-minute rate = 757115.18 events/second
    15-minute rate = 757835.58 events/second
```

Reply:
:+1: I like it. I'm not sure if it's worthwhile to limit values to compile-time constants because that prevents the toll from being used within another library, even when the cardinality is known to be low.

Reply:
Easier to open that up later than to ratchet it down, happy to merge with that constraint.